airflow/providers/amazon/aws/utils/eks_kube_config.py

```diff
@@ -0,0 +1,132 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import os
+from shutil import which
+from typing import Optional
+
+import boto3
+import yaml
+
+HOME = os.environ.get('HOME', '/tmp')
+DEFAULT_KUBE_CONFIG_FILENAME = 'config'
+DEFAULT_KUBE_CONFIG_PATH = str(os.path.join(HOME, '/.kube/', DEFAULT_KUBE_CONFIG_FILENAME))
+DEFAULT_CONTEXT_NAME = 'aws'
+DEFAULT_NAMESPACE_NAME = 'default'
+DEFAULT_POD_USERNAME = 'aws'
+
+
+def generate_config_file(
+    eks_cluster_name: str,
+    eks_namespace_name: str,
+    aws_profile: Optional[str],
+    kube_config_file_location: Optional[str] = DEFAULT_KUBE_CONFIG_PATH,
+    pod_username: Optional[str] = DEFAULT_POD_USERNAME,
+    pod_context: Optional[str] = DEFAULT_CONTEXT_NAME,
+    role_arn: Optional[str] = None,
+    aws_region: Optional[str] = None,
+) -> None:
+    """
+    Writes the kubeconfig file given an EKS Cluster name, AWS region, and file path.
+
+    :param eks_cluster_name: The name of the cluster to create the EKS Managed Nodegroup in.
+    :type eks_cluster_name: str
+    :param eks_namespace_name: The namespace to run within kubernetes.
+    :type eks_namespace_name: str
+    :param aws_profile: The named profile containing the credentials for the AWS CLI tool to use.
+    :type aws_profile: str
+    :param kube_config_file_location: Path to save the generated kube_config file to.
+    :type kube_config_file_location: str
+    :param pod_username: The username under which to execute the pod.
+    :type pod_username: str
+    :param pod_context: The name of the context access parameters to use.
+    :type pod_context: str
+    :param role_arn: The Amazon Resource Name (ARN) of the IAM role to associate with your nodegroup.
+    :type role_arn: str
+    :param aws_region: The name of the AWS Region the EKS Cluster resides in.
+    :type aws_region: str
+    """
+    installed = which("aws")
```

Comment:
How to pass the credentials to AWS CLI when we have Airflow installed at another cloud provider? In the case of ``GKEStartPodOperator``, we use [the ``GoogleBaseHook.provide_authorized_gcloud`` method](https://github.com/apache/airflow/blob/2625007c8aeca9ed98dea361ba13c2622482d71f/airflow/providers/google/common/hooks/base_google.py#L483).

Reply:
The code as submitted is working on the Breeze test environment, I don't think it should matter what's hosting it.  The line you highlighted specifically is only there as a check to make sure the AWS CLI tool is installed as that is (currently) a prerequisite.

Reply:
This is important because you probably used the default credentials from the CLI in Breeze, as you were logged in there before.

In a production environment, these credentials will be provided by Airflow e.g. via the Secret backend, So AWS CLI must be able to share the same credentials. Some Airflow users have a lot of AWS credentials for different accounts and we should be able to use them correctly in this operator as well. The user should not be aware in any way that a CLI is being used to get the token.

Reply:
Based on the above discussion, we are likely to drop the AWS CLI requirement.  If we do, does that handle the concerns here or are they still an issue?

See: https://github.com/apache/airflow/pull/16571#discussion_r659252727

Reply:
AWS CLI dependency removed in https://github.com/apache/airflow/pull/16571/commits/a6ee2c369227212e12b8d5320c91fd63a6dc769d